_posts/2022-12-21-easily-verify-your-rust-in-ci-with-kani.md

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title: "Easily verify your Rust in CI with Kani and Github Actions"
+---
+
+Formal verification tools like Kani definitively check that certain classes of bugs will not occur in your program under any circumstance.
+Kani uses mathematical techniques to explore your code over all inputs, meaning that while testing is a great way to catch the *first* bug, formally verifying your code with Kani guarantees that you've caught the *last* bug.
+To learn more about Kani, and how you can use it to prevent bugs in your Rust code, you can [read the Kani documentation](https://model-checking.github.io/kani/), or [consult the real-world examples on our blog]({{site.baseurl | prepend: site.url}}).
+
+Code, however, is seldom static.
+As code evolves, how can we ensure that it remains correct; i.e. how do we catch the *next* bug?
+Many codebases already contain test suites and other checks that can run automatically.
+Developers can use a platform like [GitHub Actions](https://github.com/features/actions) to run these tests upon every code change, as part of continuous integration, and even [ensure that the tests pass before the code can be merged](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging).
+This assures developers that a code change does not introduce a bug for all the inputs that the test suite exercises.
```

Comment:
I wouldn't use `all the inputs`. Maybe I would say that `this assures developers that a code change does not introduce a bug for inputs covered by the test suite.`

Reply:
updated